Userland/Libraries/LibGfx/FillPathImplementation.h

```diff
@@ -57,23 +57,26 @@ void fill_path(Painter& painter, Path const& path, ColorFunction color_function,
         x2 += draw_offset.x();
         if (x1 > x2)
             swap(x1, x2);
-        auto set_pixel = [&](int x, int y, Color color) {
-            painter.set_pixel(x, y, color, true);
-        };
         if constexpr (fill_path_mode == FillPathMode::AllowFloatingPoints) {
             int int_x1 = ceilf(x1);
             int int_x2 = floorf(x2);
             float left_subpixel = int_x1 - x1;
             float right_subpixel = x2 - int_x2;
-            auto left_color = color_function(IntPoint(int_x1 - 1, y) - draw_origin);
-            auto right_color = color_function(IntPoint(int_x2, y) - draw_origin);
-            set_pixel(int_x1 - 1, y, left_color.with_alpha(left_color.alpha() * left_subpixel));
-            set_pixel(int_x2, y, right_color.with_alpha(right_color.alpha() * right_subpixel));
-            for (int x = int_x1; x < int_x2; x++)
-                set_pixel(x, y, color_function(IntPoint(x, y) - draw_origin));
+            int left_subpixel_x = int_x1 - 1;
+            int right_subpixel_x = int_x2;
+            painter.draw_scanline_for_fill_path(y, left_subpixel_x, right_subpixel_x + 1, [&](int offset) {
+                int x = left_subpixel_x + offset;
+                auto color = color_function(IntPoint(x, y) - draw_origin);
+                if (x == left_subpixel_x)
+                    return color.with_alpha(color.alpha() * left_subpixel);
+                if (x == right_subpixel_x)
+                    return color.with_alpha(color.alpha() * right_subpixel);
```

Comment:
I will measure later, but I think this should be better (since the compiler is going to inline this and surely move this out the loop, so this just allows calling a cheaper set_pixel())

Reply:
I've now moved these out (as part of changing the code to use `fast_u32_fill()`).